Possible data race on pipe connection

The race detector caught the following data race:

```
==================
WARNING: DATA RACE
Read at 0x00c0420042c0 by goroutine 354:
  github.com/Microsoft/go-winio.(*win32PipeListener).listenerRoutine.func1()
      c:/gopath/src/github.com/Microsoft/go-winio/pipe.go:269 +0x42
Previous write at 0x00c0420042c0 by goroutine 311:
  github.com/Microsoft/go-winio.(*win32PipeListener).listenerRoutine()
      c:/gopath/src/github.com/Microsoft/go-winio/pipe.go:280 +0x4e7
Goroutine 354 (running) created at:
  github.com/Microsoft/go-winio.(*win32PipeListener).listenerRoutine()
      c:/gopath/src/github.com/Microsoft/go-winio/pipe.go:268 +0x2ee
Goroutine 311 (running) created at:
  github.com/Microsoft/go-winio.ListenPipe()
      c:/gopath/src/github.com/Microsoft/go-winio/pipe.go:354 +0x3c5
  github.com/fsouza/go-dockerclient.newNativeServer()
      c:/gopath/src/github.com/fsouza/go-dockerclient/client_windows_test.go:37 +0x256
  github.com/fsouza/go-dockerclient.TestPingErrorWithNativeClient()
      c:/gopath/src/github.com/fsouza/go-dockerclient/client_test.go:462 +0x73
  testing.tRunner()
      C:/go/src/testing/testing.go:747 +0x173
==================
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xd9d0bb]
```

This is the build failure: https://ci.appveyor.com/project/fsouza/go-dockerclient/build/job/xeqy446run4bfeu8

I can see two possible fixes:

1. Don't set ``p = nil`` (why is that needed?)
1. Pass a copy of p to the goroutine that invokes ``connectPipe``

--- a/pipe.go
+++ b/pipe.go
@@ -265,9 +265,9 @@ func (l *win32PipeListener) listenerRoutine() {
 			if err == nil {
 				// Wait for the client to connect.
 				ch := make(chan error)
-				go func() {
+				go func(p *win32File) {
 					ch <- connectPipe(p)
-				}()
+				}(p)
 				select {
 				case err = <-ch:
 					if err != nil {

